user-journey-tests/helpers/testHelpers/biosecurity.js

```diff
@@ -19,31 +19,33 @@ const completeBiosecurityTask = async (radioType) => {
   await taskListPage.selectBiosecurityLink()
   switch (radioType) {
     case 'yes':
-      await keptSeparatelyPage.selectYesAndContinue()
-      await waitForPagePath(grazingPage.pagePath)
-      await grazingPage.selectYesAndContinue()
-      await waitForPagePath(lastGrazedPage.pagePath)
-      await lastGrazedPage.inputLastGrazedAndContinue('2 years')
-      await waitForPagePath(manureAndSlurryPage.pagePath)
-      await manureAndSlurryPage.selectYesAndContinue()
-      await waitForPagePath(howFieldSeparatedPage.pagePath)
+      await keptSeparatelyPage.selectYesAndContinue(grazingPage)
+      await grazingPage.selectYesAndContinue(lastGrazedPage)
+      await lastGrazedPage.inputLastGrazedAndContinue(
+        '2 years',
+        manureAndSlurryPage
+      )
+      await manureAndSlurryPage.selectYesAndContinue(howFieldSeparatedPage)
       await howFieldSeparatedPage.inputSeparatedGrazingAndContinue(
-        'Separate grazing'
+        'Separate grazing',
+        roadsAndTracksPage
+      )
+      await roadsAndTracksPage.selectYesAndContinue(anySharedBuildingsPage)
+      await anySharedBuildingsPage.selectYesAndContinue(
+        minimiseContaminationPage
       )
-      await waitForPagePath(roadsAndTracksPage.pagePath)
-      await roadsAndTracksPage.selectYesAndContinue()
-      await waitForPagePath(anySharedBuildingsPage.pagePath)
-      await anySharedBuildingsPage.selectYesAndContinue()
-      await waitForPagePath(minimiseContaminationPage.pagePath)
       await minimiseContaminationPage.inputMinimiseContaminationAndContinue(
-        'Minimise'
+        'Minimise',
+        disinfectionPage
+      )
+      await disinfectionPage.inputPeopleDisinfectionAndContinue(
+        'People disinfection',
+        disinfectantPage
       )
-      await waitForPagePath(disinfectionPage.pagePath)
       break
 
     case 'no':
-      await keptSeparatelyPage.selectNoAndContinue()
-      await waitForPagePath(disinfectionPage.pagePath)
+      await keptSeparatelyPage.selectNoAndContinue(disinfectionPage)
```

Comment:
With the new implementation I would add ToPage to the function name to make it clear that it checks the next page